File: 0014-longest-common-prefix.rb
# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
  for i in 0..200 do
    matches = true
    for str in strs
      if str[i] != strs[0][i] then
        matches = false
        break
      end
    end
    if not matches then
      break
    end
  end

  if i > 0 then
    prefix = strs[0][0..i-1]
  else
    prefix = ""
  end
  return prefix
end

puts longest_common_prefix(["flower", "flow", "flight"])
puts longest_common_prefix(["dog", "racecar", "car"])
puts longest_common_prefix(["dog", "dogeball"])

File: 0014-longest-common-prefix.ts
function longestCommonPrefix(strs: string[]): string {
    let matches = false;
    let i = 0;
    while (i < 200) {
        matches = true;
        for (let str of strs) {
            if (str[i] != strs[0][i]) {
                matches = false;
            }
        }
        if (!matches) {
            break;
        }
        i++;
    }

    let prefix: string;
    if (i > 0) prefix = strs[0].substring(0, i);
    else prefix = "";
    return prefix;
}

console.log(longestCommonPrefix(["flower", "flow", "flight"]));
console.log(longestCommonPrefix(["dog", "racecar", "car"]));
console.log(longestCommonPrefix(["dog", "dogeball"]));
